### PR TITLE
guard against no L1T uGT digis in `L1TriggerResultsConverter`

### LINKED ISSUE #40494: Failure in 2022 data promptNano (L1)
We have spotted the following failure in the 2022 data promptNano production, presumably connected to L1 trigger:

https://cms-unified.web.cern.ch/cms-unified/report/cmsunified_ACDC0_Run2022G_MinimumBias_PromptNanoAODv10_v1_221229_092537_1876

The crash info:

```
Fatal Exception (Exit code: 8002) 
An exception of category 'StdException' occurred while
[0] Processing Event run: 362859 lumi: 1 event: 8887 stream: 1
[1] Running path 'NANOEDMAODoutput_step'
[2] Prefetching for module PoolOutputModule/'NANOEDMAODoutput'
[3] Calling method for module L1TriggerResultsConverter/'l1bits'
Exception Message:
A std::exception was thrown.
vector::_M_range_check: __n (which is 0) >= this->size() (which is 0)
```

The problem is reproducible at different sites.

PdmV

### PR DESCRIPTION
**Pull request by missirol**
#### PR description:

This PR fixes the edge case reported in #40494. It adds a check on the size of the L1T uGT digis, returning `false` for all L1T algorithms when said input collection has `0` elements for the relevant BX (see https://github.com/cms-sw/cmssw/pull/40528#discussion_r1081662085 for details).

#### PR validation:

Fixes the crash reproduced from the instructions in #40494 (tested in `13_0_X`):
```
cmsDriver.py RECO --conditions 124X_dataRun3_Prompt_v4 --customise Configuration/DataProcessing/Utils.addMonitoring \
--datatier NANOAOD --era Run3 --eventcontent NANOEDMAOD \
--filein /store/data/Run2022G/MinimumBias/MINIAOD/PromptReco-v1/000/362/859/00000/948464c4-75ad-41a3-95d0-82e31c865b10.root \
--fileout "file:ReReco-Run2022G-MinimumBias-PromptNanoAODv10_v1-00001.root" --nThreads 1 --number -1 \
--python_filename "ReReco-Run2022G-MinimumBias-PromptNanoAODv10_v1-00001_0_cfg.py" --scenario pp --step NANO --data
```

#### If this PR is a backport, please specify the original PR and why you need to backport that PR. If this PR will be backported, please specify to which release cycle the backport is meant for:

TBD

### CODE REVIEW

**Comment by missirol** [comment]
type bugfix

**Comment by missirol** [comment]
@cms-sw/xpog-l2 

Can you please review this PR?

**Comment by vlimant** [comment]
see comments above

**Comment by missirol** [comment]
there are no comments, maybe you didnt hit 'finish review'

File: PhysicsTools/NanoAOD/plugins/L1TriggerResultsConverter.cc [hunk 1]
```diff
@@ -125,16 +125,17 @@ void L1TriggerResultsConverter::beginRun(edm::Run const&, edm::EventSetup const&
 // ------------ method called to produce the data  ------------
 
 void L1TriggerResultsConverter::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
-  using namespace edm;
   const std::vector<bool>* wordp = nullptr;
   bool unprefireable_bit = false;
   if (!legacyL1_) {
     const auto& resultsProd = iEvent.get(token_);
-    wordp = &resultsProd.at(0, 0).getAlgoDecisionFinal();
+    if (resultsProd.size() > 0) {
```
[the remark this answers is not in the document]
Comment (missirol) [reply PhysicsTools/NanoAOD/plugins/L1TriggerResultsConverter.cc]: Good point.. I don't know (maybe @cms-sw/l1-l2 knows?), and I had the same doubt.

I used `size` to be consistent with
https://github.com/cms-sw/cmssw/blob/df24460a422a36e623e3ac5c0d4a4672ebff1773/PhysicsTools/NanoAOD/plugins/L1TriggerResultsConverter.cc#L137
and similar cases seen elsewhere
https://github.com/cms-sw/cmssw/blob/73157686b53289c96d5b31b086987b854ac70dc5/HLTrigger/HLTfilters/plugins/HLTL1TSeed.cc#L479
but now I also see cases where the `0` is explicitly used
https://github.com/cms-sw/cmssw/blob/73157686b53289c96d5b31b086987b854ac70dc5/DQM/HLTEvF/plugins/TriggerRatesMonitor.cc#L361

I think the correct call is `size(0)` (actually, `isEmpty(0)` is sufficient and a bit cheaper), so I switched to that, for both `resultsProd` and `handleExtResults`.

**Comment by missirol** [comment]
please test

**Comment by missirol** [comment]
@cms-sw/xpog-l2 

Do you have other comments on this PR?

**Comment by vlimant** [comment]
+1

**Comment by missirol** [comment]
> a backport might be required, although data taking will move on to 13.0 AFAIK

The 12_6_X backport is in place, and tested, at https://github.com/cms-sw/cmssw/pull/40588.

**Comment by perrotta** [comment]
+1

**Pull request merged**